test/test_updst.cxx

```diff
@@ -37,8 +37,12 @@ TEST(ecmech, updst_a)
 
 #if KIN_TYPE
    matModelEvptn_FCC_B* mmodel = new matModelEvptn_FCC_B();
+   constexpr int nH = matModelEvptn_FCC_B::nH;
```

Comment:
So, the issue was in our debug builds in which things would break (at least with `rocm v4.x`) as the HIP compiler would not accept things like the below. So, I had to introduce the constexpr of things to get around that issue. I can take a look again and see if the newer versions of rocm accept the earlier code, and if not I'll look at a cleaner workaround with less duplicated code.

``` c++
   // Compiler would error out as it did not recognize the mmodel->nH term as being a constexpr term
   // So, we had to explicitly move to the lower version of things to get things to work...
   // Also worth noting is that none of our other (host or GPU) compilers complained about this before 
   // printVec<mmodel->nH>(&(hist[ecmech::evptn::iHistLbH]), std::cout);
   ecmech::printVec<nH>(&(hist[ecmech::evptn::iHistLbH]), std::cout);
```

Reply:
ROCm still can't deal with the earlier mentioned case. However, I simplified the logic around this code so it should make things simpler going forward.